### PR TITLE
Refactor Starwars tutorial haml to be more generalizable

### PR DESCRIPTION
**Pull request by Erin007**
On our tutorial landing pages we have 3 very similar components that are all currently using [Starwars styling and selectors](https://github.com/code-dot-org/code-dot-org/pull/25528/files#r228708365) since they were built quickly to hit HoC deadlines: 

**Starwars**
screen shot 2018-11-30 at 5 32 26 pm

**Minecraft** 
screen shot 2018-11-30 at 5 33 21 pm

**Dance** 
screen shot 2018-11-30 at 5 32 48 pm

It's also very likely that we will want to use a similar component in the future. 

This PR is the first in a series to extract the skeleton of the Starwars version of the component and its stylings into a more general, reusable haml partial. This PR affects only code.org/starwars. Follow up PR will address the other 2 components and delete the styles duplication. 

There should be no user impact with this refactor, and there are eyes tests on the Starwars landing page, which should catch any regressions. 

English // Desktop // Before 
english desktop prod

English // Desktop // After 
english desktop after

English // Mobile // Before 
english mobile before

English // Mobile // After 
english mobile after

non-English // Desktop // Before 
spanish desktop prod

non-English // Desktop // After 
spanish desktop after

non-English // Mobile // Before 
spanish mobile before

non-English // Mobile // After 
spanish mobile after

### CODE REVIEW

File: pegasus/sites.v3/code.org/public/css/tutorial-promo.css [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  .tutorial-heading {
+    top: 15px;
+    left: 290px;
+  }
+
+  .img-container {
+    padding: 0px 5px;
+  }
+
+  #starwars-mobile-image {
```
Comment (breville) [inline pegasus/sites.v3/code.org/public/css/tutorial-promo.css]: Should we find a way to get this one "starwars" reference out of this file?

File: pegasus/sites.v3/code.org/views/tutorial_promo.haml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    %img.background-img{src: tutorial[:background_image]}
+  %h1.tutorial-heading=tutorial[:heading]
+
+  .col-25.specific-tutorial.left-tutorial{style: "display: #{tutorial[:display_second_tutorial]}"}
+    .tutorial-box{id: tutorial[:id_for_mobile_image]}
+      %h2.specific-tutorial-heading=tutorial[:left_heading]
+      .tutorial-info
+        %p.tutorial-description=tutorial[:left_description]
+        %p.tutorial-specs=tutorial[:left_specs]
+      %a{:href => tutorial[:left_button_link], :target=>'_self'}
```
Comment (breville) [inline pegasus/sites.v3/code.org/views/tutorial_promo.haml]: Can we update from `=>` to `:` syntax?

File: pegasus/sites.v3/code.org/public/starwars.haml [hunk 3]
```diff
@@ -23,30 +23,27 @@ style_min: true
...
 %link{href: "/css/starwars.css", rel: "stylesheet"}
 
-%div{style: "clear:both"}
-
-.starwars-container
-  .img-container
-    %img.starwars-img{src:"/images/fit-1940/sw_landingpage.jpg"}
-  %h1.starwars-heading= I18n.t(:starwars_subtitle)
+:ruby
+  display_second_tutorial = !("#{request.locale}" == "en-US") ? 'none' : ''
```
Comment (nkiruka) [inline pegasus/sites.v3/code.org/public/starwars.haml:27]: Nice cleanup!

File: pegasus/sites.v3/code.org/public/starwars.haml [hunk 4]
```diff
@@ -23,30 +23,27 @@ style_min: true
...
 
-  .col-25.starwars-tutorial.starwars-blocks-tutorial
-    .tutorial-box
-      %h2.sw-h2= I18n.t(:starwars_blocks_title)
-      .sw-tutorial-info
-        %p.sw-tutorial-description= I18n.t(:starwars_blocks_description)
-        %p.sw-tutorial-specs= I18n.t(:starwars_blocks_specs)
-      %a{:href => '/api/hour/begin/starwarsblocks', :target=>'_self'}
-        %button.sw-try-button= I18n.t(:try_now)
+=view :tutorial_promo, tutorial: tutorial
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/public/starwars.haml]: nit: add a space
```suggestion
= view :tutorial_promo, tutorial: tutorial
```

File: pegasus/sites.v3/code.org/views/tutorial_promo.haml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+%link{href: "/css/tutorial-promo.css", rel: "stylesheet"}
+
+.tutorial-promo-container
+  .img-container
+    %img.background-img{src: tutorial[:background_image]}
+  %h1.tutorial-heading=tutorial[:heading]
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/views/tutorial_promo.haml]:
```suggestion
  %h1.tutorial-heading= tutorial[:heading]
```

File: pegasus/sites.v3/code.org/views/tutorial_promo.haml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  .col-25.specific-tutorial.left-tutorial{style: "display: #{tutorial[:display_second_tutorial]}"}
+    .tutorial-box{id: tutorial[:id_for_mobile_image]}
+      %h2.specific-tutorial-heading=tutorial[:left_heading]
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/views/tutorial_promo.haml]:
```suggestion
      %h2.specific-tutorial-heading= tutorial[:left_heading]
```

File: pegasus/sites.v3/code.org/views/tutorial_promo.haml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      .tutorial-info
+        %p.tutorial-description=tutorial[:left_description]
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/views/tutorial_promo.haml]:
```suggestion
        %p.tutorial-description= tutorial[:left_description]
```

File: pegasus/sites.v3/code.org/views/tutorial_promo.haml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+        %p.tutorial-specs=tutorial[:left_specs]
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/views/tutorial_promo.haml]:
```suggestion
        %p.tutorial-specs= tutorial[:left_specs]
```

File: pegasus/sites.v3/code.org/views/tutorial_promo.haml [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+      %a{:href => tutorial[:left_button_link], :target=>'_self'}
+        %button.tutorial-button= tutorial[:left_button_text]
+
+  .col-25.specific-tutorial.right-tutorial
+    .tutorial-box{id: tutorial[:id_for_mobile_image]}
+      %h2.specific-tutorial-heading=tutorial[:right_heading]
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/views/tutorial_promo.haml]:
```suggestion
      %h2.specific-tutorial-heading= tutorial[:right_heading]
```

File: pegasus/sites.v3/code.org/views/tutorial_promo.haml [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+      .tutorial-info
+        %p.tutorial-description=tutorial[:right_description]
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/views/tutorial_promo.haml]:
```suggestion
        %p.tutorial-description= tutorial[:right_description]
```

File: pegasus/sites.v3/code.org/views/tutorial_promo.haml [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+        %p.tutorial-specs=tutorial[:right_specs]
```
Comment (maddiedierker) [inline pegasus/sites.v3/code.org/views/tutorial_promo.haml]:
```suggestion
        %p.tutorial-specs= tutorial[:right_specs]
```

**Review by maddiedierker — APPROVED** [review]
sorry for all the nit comments! this is great clean-up 🥇

**Review by breville — COMMENTED** [review]
Great work!

**Pull request merged**